test/functional/orphaned-renames/test.bats

```diff
@@ -46,16 +60,35 @@ setup() {
 
   sudo $CREATE_UPDATE --osversion 30 --statedir $DIR --format 3
   sudo $MAKE_FULLFILES --statedir $DIR 30
-  # version 10: add file to 10
-  [ 1 -eq $(grep 'F\.\.\.	.*	10	/usr/lib/bar' $DIR/www/10/Manifest.test-bundle | wc -l) ]
-  # version 20: rename bar to baz
-  [ 1 -eq $(grep '\.d\.r	.*	20	/usr/lib/bar' $DIR/www/20/Manifest.test-bundle | wc -l) ]
-  [ 1 -eq $(grep 'F\.\.r	.*	20	/usr/lib/baz' $DIR/www/20/Manifest.test-bundle | wc -l) ]
+
+  set_latest_ver 30
+
+  sudo $CREATE_UPDATE --osversion 40 --statedir $DIR --format 3
+  sudo $MAKE_FULLFILES --statedir $DIR 40
+  # version 10: add files to 10
+  [ 1 -eq $(grep $'F\.\.\.\t.*\t10\t/usr/lib/bar' $DIR/www/10/Manifest.test-bundle | wc -l) ]
+  [ 1 -eq $(grep $'F\.\.\.\t.*\t10\t/one' $DIR/www/10/Manifest.test-bundle | wc -l) ]
+  # version 20: rename bar to baz and one to two
+  [ 1 -eq $(grep $'\.d\.r\t.*\t20\t/one' $DIR/www/20/Manifest.test-bundle | wc -l) ]
+  [ 1 -eq $(grep $'F\.\.r\t.*\t20\t/two' $DIR/www/20/Manifest.test-bundle | wc -l) ]
+  [ 1 -eq $(grep $'\.d\.r\t.*\t20\t/usr/lib/bar' $DIR/www/20/Manifest.test-bundle | wc -l) ]
+  [ 1 -eq $(grep $'F\.\.r\t.*\t20\t/usr/lib/baz' $DIR/www/20/Manifest.test-bundle | wc -l) ]
   # version 30: prune original renamed-from file (bar), baz is now a renamed-from file
   # Check for the new renamed-to file (foo)
-  [ 0 -eq $(grep '/usr/lib/bar' $DIR/www/30/Manifest.test-bundle | wc -l) ]
-  [ 1 -eq $(grep '\.d\.r	.*	30	/usr/lib/baz' $DIR/www/30/Manifest.test-bundle | wc -l) ]
-  [ 1 -eq $(grep 'F\.\.r	.*	30	/usr/lib/foo' $DIR/www/30/Manifest.test-bundle | wc -l) ]
+  # /two was deleted in this version, so both /one and /two should be marked as deleted
+  [ 1 -eq $(grep $'\.d\.\.\t0000000000000000000000000000000000000000000000000000000000000000\t20\t/one' $DIR/www/30/Manifest.test-bundle | wc -l) ]
```

Comment:
Not sure if other tests write out the zeros like above, but you can make the regexp shorter. For example: `\.d\.\.\t0\{64\}\t20\t/one`.

Reply:
right, ok.